elm-sdks/morphir-unit-test/src/Morphir/UnitTest/Expect.elm

```diff
@@ -0,0 +1,120 @@
+module Morphir.UnitTest.Expect exposing(..)
+import Morphir.SDK.Dict exposing(Dict, toList, get)
+import Morphir.SDK.List exposing(List)
+import Morphir.SDK.Set exposing(Set)
+
+type Expectation = Expectation ExpectationResult
+type ExpectationResult = Pass | Fail String
+
+getResult : Expectation -> ExpectationResult
+getResult Expectation result = result
+
+getResultBool : Expectation -> Bool
+getResultBool e = case e of
+    Expectation Pass -> True
+    Expectation (Fail _) -> False
+
+equal : a -> a -> Expectation
+equal x y = if (x == y)
+    then Expectation Pass
+    else Expectation (Fail "Values not Equal")
+
+notEqual : a -> a -> Expectation
+notEqual x y = if (x /= y)
+    then Expectation Pass
+    else Expectation (Fail "Values are Equal")
+
+all : List (subject -> Expectation) -> subject -> Expectation
+all expectations subject = 
+    let 
+        allPassed = 
+            List.all
+                (\f -> case (f subject) of
+                    Expectation (Fail _) -> False
+                    _ -> True
+                )
+            expectations 
+    in
+    if allPassed
+    then Expectation Pass
+    else Expectation (Fail "One or more expectations failed")
+
+assert : Bool -> Expectation
+assert a = if a
+    then Expectation Pass
+    else Expectation (Fail "Assertion Failed")
+
+lessThan : comparable -> comparable -> Expectation
+lessThan a b = if (a < b)
+    then Expectation Pass
+    else Expectation (Fail "Value not less than")
+
+greaterThan : comparable -> comparable -> Expectation
+greaterThan a b = if (a > b)
+    then Expectation Pass
+    else Expectation (Fail "Value not greater than")
+
+atMost : comparable -> comparable -> Expectation
+atMost a b = if (a <= b)
+    then Expectation Pass
+    else Expectation (Fail "Value not at most")
+
+atLeast : comparable -> comparable -> Expectation
+atLeast a b = if (a >= b)
+    then Expectation Pass
+    else Expectation (Fail "Value not at least")
+
+okay : Result a b -> Expectation
+okay a = case a of
+    Ok _ -> Expectation Pass
+    Err _ -> Expectation (Fail "Result is an error")
+
+err : Result a b -> Expectation
+err a = case a of
+    Ok _ -> Expectation (Fail "Expected Error")
+    Err _ -> Expectation Pass
+
+equalLists : List a -> List a -> Expectation
+equalLists l1 l2 = if (l1 == l2)
+    then Expectation Pass
+    else Expectation (Fail "Lists not Equal")
+
+
+equalDicts : Dict comparable a -> Dict comparable a -> Expectation
+equalDicts d1 d2 = 
+    let l1 = toList d1 in
+    let l2 = toList d2 in
+    let 
+        l1_matches = List.all 
+            (\(k1, v1) -> 
+                case get k1 d2 of
+                    Just v2 -> v1 == v2
+                    Nothing -> False) 
+         l1 
+    in
+    let 
+        l2_matches = List.all 
```

Comment:
It seems inefficient to check all these if you've already determined that `l1_matches` is false. Also, comparing the length of the two lists before checking for matches seems like it might be more efficient.

Reply:
Good point, changed to just a list length check for one of them.